### PR TITLE
Hash page row

### PR DESCRIPTION
**Pull request by keith-turner**
I ran some test on EC2 using roughly the same config as the [recent long test][1].  The following plot shows loading 80 common crawl files.  Before this 120 common crawl files were loaded using spark.  The dips in performance are caused by files finishing and new ones starting.  With the changes in this PR, each file ran as a task.  There were 20 spark executors for loading data.  The load across the cluster was much more even than in the past and the throughput was better.

hash-run

Below are some the tablets in the Fluo table containing page data.  The file sizes are very even.  This was not the case before adding the hash.

```
5;p:0aw3 file:hdfs://leader1:10000/accumulo/tables/5/t-0000279/A0000dud.rf []    41977575,126316
5;p:0ls6 file:hdfs://leader1:10000/accumulo/tables/5/t-000027r/A0000due.rf []    42176089,126148
5;p:0wo9 file:hdfs://leader1:10000/accumulo/tables/5/t-00001ne/A0000duf.rf []    42035837,125046
5;p:17kc file:hdfs://leader1:10000/accumulo/tables/5/t-0000278/A0000dug.rf []    41237240,125422
5;p:1igf file:hdfs://leader1:10000/accumulo/tables/5/t-000027q/A0000duh.rf []    41937943,125324
5;p:1tci file:hdfs://leader1:10000/accumulo/tables/5/t-00001jq/A0000dui.rf []    41119372,124920
5;p:248l file:hdfs://leader1:10000/accumulo/tables/5/t-0000274/A0000dcd.rf []    41955329,125664
5;p:2f4o file:hdfs://leader1:10000/accumulo/tables/5/t-000027m/A0000dce.rf []    41337171,124974
5;p:2q0r file:hdfs://leader1:10000/accumulo/tables/5/t-00001nc/A0000dcf.rf []    41315244,125138
```

[1]: http://fluo.io/webindex-long-run

### CODE REVIEW

**Comment by keith-turner** [comment]
This PR depends on https://github.com/fluo-io/fluo-recipes/pull/52

File: bin/webindex [hunk 1]
```diff
@@ -149,16 +149,6 @@ load-s3)
     ${COMMAND}
   fi
   ;;
-reindex)
```
Comment (mikewalch) [inline bin/webindex]: While I think it's OK that the reindex command is removed for this PR, I think it is something we should tackle.  I think it's a problem that all production Fluo applications will need to solve and we should work through it to determine if new features or recipes are needed and show that is possible.  What do you think about adding this as an issue?

File: modules/data/src/main/java/io/fluo/webindex/data/spark/AccumuloRangePartitioner.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import io.fluo.api.data.Bytes;
+import io.fluo.api.data.RowColumn;
+import org.apache.hadoop.io.Text;
+import org.apache.spark.Partitioner;
+
+public class AccumuloRangePartitioner extends Partitioner {
```
Comment (mikewalch) [inline modules/data/src/main/java/io/fluo/webindex/data/spark/AccumuloRangePartitioner.java]: Is this something that should eventually be added to recipes?  If so, might be good to create an issue.

Comment (keith-turner) [inline modules/data/src/main/java/io/fluo/webindex/data/spark/AccumuloRangePartitioner.java]: Maybe, its more general than Fluo.  Its more of an Accumulo+Spark integration thing. There is no other good place to put it I suppose.  Would be nice to be able to easily reuse it.

**Comment by mikewalch** [comment]
+1

**Pull request merged**